cdap-ui/app/cdap/components/ResourceCenterButton/index.js

```diff
@@ -19,14 +19,29 @@ import React, { Component } from 'react';
 import PlusButtonStore from 'services/PlusButtonStore';
 import PlusButtonModal from 'components/PlusButtonModal';
 import classnames from 'classnames';
+import ee from 'event-emitter';
+import globalEvents from 'services/global-events';
 require('./ResourceCenterButton.scss');
 
+/**
+ * This module is used solely for angular side. This includes styling and stripping of context
+ * information used by the PlusButton. We may nee to consider using one component for both angular
+ * and react as it seems redundant.
+ */
 export default class ResourceCenterButton extends Component {
+  eventemitter = ee(ee);
+  onClickHandler = () => {
```

Comment:
I think this will be better to be set with 2 different explicit functions (just in case there's race condition where multiple events firing?).

Also nit to move these under the lifecycle functions

Reply:
Not sure setting as two different modals will resolve the problem of race condition though. If the event gets fired in cycles which causes the modal to be closed and reopened it will happen no matter what function handles it right? Am I missing something here?

Move under lifecycle functions as in the click handler? or the event registration?

Reply:
I mean if the CLOSE event gets fired twice, then it will actually open the modal again with the current implementation.

The Click handler location in the file should be under the react lifecycle functions per codestyle

Reply:
oh! you mean the placement of the click handlers! ok Have updated them + separated out event handlers.